packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -185,15 +199,15 @@ function useFocusContainment(scopeRef: RefObject<HTMLElement[]>, contain: boolea
   let focusedNode = useRef<HTMLElement>();
 
   let raf = useRef(null);
-  useEffect(() => {
+  useLayoutEffect(() => {
```

Comment:
Needed so the focus event is registered before elements within the scope are auto focused, or a parent scope will end up handling the event instead and restoring focus back to that scope.